staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -398,6 +399,7 @@ func (pf *PortForwarder) handleConnection(conn net.Conn, port ForwardedPort) {
 	err = <-errorChan
 	if err != nil {
 		runtime.HandleError(err)
+		os.Exit(1)
```

Comment:
it isn't appropriate to exit from library code.  It is possible to add ErrorHandlers in your binary if you want different behavior.  I'm also ok returning an error from this method and up the chain so that the caller can decide how to handle the situation.

Reply:
Hey @deads2k /me nodes. Let me work in the up the chain way.